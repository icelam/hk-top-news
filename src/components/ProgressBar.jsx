import React from 'react';

/* Material UI Components */
import LinearProgress from '@material-ui/core/LinearProgress';

const ProgressBar = () => <LinearProgress color="secondary" />;

export default ProgressBar;
