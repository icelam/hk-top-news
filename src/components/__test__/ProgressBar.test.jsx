import React from 'react';
import { render } from '@testing-library/react';
import ProgressBar from '../ProgressBar';

describe('Progress Bar', () => {
  it('renders correctly', () => {
    const { container, asFragment } = render(<ProgressBar data-testid="progress-bar" />);
    expect(container.firstChild).toHaveClass('MuiLinearProgress-root');
    expect(container.querySelector('.MuiLinearProgress-bar')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
