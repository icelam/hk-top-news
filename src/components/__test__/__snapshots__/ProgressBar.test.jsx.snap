// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Bar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Indeterminate"
    />
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorSecondary"
    />
  </div>
</DocumentFragment>
`;
